src/main/java/aegis/Aegis.java

```diff
@@ -0,0 +1,282 @@
+            System.out.printf(" New event added: %n   %s%n", newEvent);
+            System.out.printf(" %d tasks needed to be done. Let me assist you!%n", taskList.size());
+        } catch (AegisException e) {
+            System.out.printf(" Anomaly detected: [%s]%n", e.getMessage());
+        }
+    }
+
+    private static void addTodoTask(ArrayList<Task> taskList, String input) {
+        try {
+            String description = input.replaceFirst("todo ", "").trim();
+            if (description.isEmpty()) {
+                throw new AegisException("The description of a todo cannot be empty");
+            }
+            Todo newTodo = new Todo(description);
+            taskList.add(newTodo);
+            saveTasksToFile(taskList);
+            System.out.printf(" New todo added: %n   %s%n", newTodo);
+            System.out.printf(" %d tasks needed to be done. Let me assist you!%n", taskList.size());
+        } catch (AegisException e) {
+            System.out.printf(" Anomaly detected: [%s]%n", e.getMessage());
+        }
+    }
+
+    private static void deleteTask(ArrayList<Task> taskList, String input) {
+        try {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+            if (taskIndex < 0 || taskIndex >= taskList.size()) {
+                throw new AegisException("Invalid task number");
+            }
+            Task removedTask = taskList.remove(taskIndex);
+            saveTasksToFile(taskList);
+            System.out.printf(" Noted. I've removed this task:%n   %s%n", removedTask);
+            System.out.printf(" Now you have %d tasks in the list.%n", taskList.size());
+        } catch (AegisException | NumberFormatException e) {
+            System.out.printf(" Anomaly detected: [%s]%n", e.getMessage());
+        }
+    }
+
+    private static void saveTasksToFile(ArrayList<Task> taskList) {
+        try {
+            File file = new File(FILE_PATH);
+            file.getParentFile().mkdirs();
+            FileWriter writer = new FileWriter(file);
+            for (Task task : taskList) {
+                writer.write(task.toFileFormat() + System.lineSeparator());
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println(" Anomaly detected: Error saving tasks to file.");
+        }
+    }
+
+    private static void loadTasksFromFile(ArrayList<Task> taskList) {
+        try {
+            File file = new File(FILE_PATH);
+            if (!file.exists()) {
+                System.out.println(" No previous task list found. Starting fresh.");
+                return;
+            }
+
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(" \\| ");
+                String taskType = parts[0];
+                boolean isDone = parts[1].equals("1");
+
+                switch (taskType) {
```

Comment:
Try to avoid magic literals. You can consider using named constants, especially when it's used multiple times in the code.